azure-iot-device/azure/iot/device/iothub/sync_clients.py

```diff
@@ -160,6 +161,22 @@ def _generic_receive_handler_setter(self, handler_name, feature_name, new_handle
         elif new_handler is None and self._mqtt_pipeline.feature_enabled[feature_name]:
             self._disable_feature(feature_name)
 
+    @staticmethod
+    def _generate_pnp_handler_translation_wrapper(handler_to_wrap, translation_fn):
+        """Generate a translation wrapper for a PNP-related handler, using the given
+        translation function.
+
+        :param handler_to_wrap: Handler function that will be wrapped
+        :param translation_fn: Function that translates a non-PNP object passed by the handler
+            into the PNP equivalent
+        """
+
+        def translation_wrapper(handler_obj):
+            translated_obj = translation_fn(handler_obj)
+            handler_to_wrap(translated_obj)
```

Comment:
>handler_to_wrap(translated_obj) [](http://example.com/codeflow?start=12&length=31)

Do handlers ever return values?  I think the answer is no.  If the answer is yes, then you need to add `return` here.  Or just add it either way because it doesn't hurt.

Reply:
Handlers never return values in our clients, and unless something massive were to change internally, they never will.